src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionDiagnosticAnalyzer.cs

```diff
@@ -358,6 +360,39 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext, INamedTyp
             return false;
         }
 
+        private static bool CanReplaceDelegateWithLocalFunction(
+            INamedTypeSymbol delegateType, 
+            LocalDeclarationStatementSyntax localDeclaration,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            var delegateTypeParamNames = delegateType.GetAllTypeParameters().Select(p => p.Name).ToImmutableHashSet();
```

Comment:
❓ Can we fast-path this for the case where the delegate is not contained within a generic type? I believe that would cover the vast majority of delegates, so I'm envisioning this:

```csharp
var delegateContainingType = delegateType.ContainingType;
if (delegateContainingType is null || !delegateContainingType.IsGenericType)
{
    return true;
}
```

⚠️ This is very different from checking if the delegate itself is a generic type, which is quite common.

📝 Due to the age of this pull request, I am only looking for an answer to the question. I plan to update the pull request myself to prepare it for merge.